### ISSUE TITLE
Installation error: network connect ETIMEDOUT

### ISSUE THREAD

**Issue by dathanasakoglou** [Glavin001/atom-beautify#1462]
Installing “atom-beautify@0.29.16” failed.
INFO: Could not find files for the given pattern(s).

The system cannot find the file specified.

npm ERR! Windows_NT 10.0.14393
npm ERR! argv "C:\\Users\\dimit\\AppData\\Local\\atom\\app-1.12.9\\resources\\app\\apm\\bin\\node.exe" "C:\\Users\\dimit\\AppData\\Local\\atom\\app-1.12.9\\resources\\app\\apm\\node_modules\\npm\\bin\\npm-cli.js" "--globalconfig" "C:\\Users\\dimit\\.atom\\.apm\\.apmrc" "--userconfig" "C:\\Users\\dimit\\.atom\\.apmrc" "install" "C:\\Users\\dimit\\AppData\\Local\\Temp\\d-117011-7580-1gfxhtj\\package.tgz" "--runtime=electron" "--target=1.3.13" "--arch=ia32" "--global-style"
npm ERR! node v4.4.5
npm ERR! npm  v3.10.5
npm ERR! code ETIMEDOUT
npm ERR! errno ETIMEDOUT
npm ERR! syscall connect

npm ERR! network connect ETIMEDOUT 151.101.112.162:443
npm ERR! network This is most likely not a problem with npm itself
npm ERR! network and is related to network connectivity.
npm ERR! network In most cases you are behind a proxy or have bad network settings.
npm ERR! network 
npm ERR! network If you are behind a proxy, please make sure that the
npm ERR! network 'proxy' config is set properly.  See: 'npm help config'

npm ERR! Please include the following file with any support request:
npm ERR!     C:\Users\dimit\AppData\Local\Temp\apm-install-dir-117011-7580-17k2hqd\npm-debug.log

running apm install atom-beautify on cmd gives me the same error.
im not on a proxy.

**Comment by BlackSkorpio** [comment]
Having a similar issue
```
Updating to “atom-beautify@0.29.16” failed.Hide output…

> atom-beautify@0.29.16 postinstall C:\Users\BJARNE~1\AppData\Local\Temp\apm-install-dir-117011-11520-wea9ix\node_modules\atom-beautify
> npm run build-options

C:\Users\BJARNE~1\AppData\Local\Temp\apm-install-dir-117011-11520-wea9ix
`-- (empty)


npm WARN deprecated node-uuid@1.4.7: use uuid module instead
npm WARN deprecated minimatch@0.2.12: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated node-uuid@1.4.0: use uuid module instead
npm WARN deprecated minimatch@2.0.10: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
C:\Users\Bjarne �r inte ett internt kommando, externt kommando,
program eller kommandofil.
npm WARN enoent ENOENT: no such file or directory, open 'C:\Users\BJARNE~1\AppData\Local\Temp\apm-install-dir-117011-11520-wea9ix\package.json'
npm WARN apm-install-dir-117011-11520-wea9ix No description
npm WARN apm-install-dir-117011-11520-wea9ix No repository field.
npm WARN apm-install-dir-117011-11520-wea9ix No README data
npm WARN apm-install-dir-117011-11520-wea9ix No license field.
npm ERR! Windows_NT 6.1.7601
npm ERR! argv "C:\\Users\\Bjarne Varöystrand\\AppData\\Local\\atom\\app-1.13.0\\resources\\app\\apm\\bin\\node.exe" "C:\\Users\\Bjarne Varöystrand\\AppData\\Local\\atom\\app-1.13.0\\resources\\app\\apm\\node_modules\\npm\\bin\\npm-cli.js" "--globalconfig" "C:\\Users\\Bjarne Varöystrand\\.atom\\.apm\\.apmrc" "--userconfig" "C:\\Users\\Bjarne Varöystrand\\.atom\\.apmrc" "install" "C:\\Users\\BJARNE~1\\AppData\\Local\\Temp\\d-117011-11520-1g4516r\\package.tgz" "--runtime=electron" "--target=1.3.13" "--arch=ia32" "--global-style"
npm ERR! node v4.4.5
npm ERR! npm  v3.10.5
npm ERR! code ELIFECYCLE

npm ERR! atom-beautify@0.29.16 postinstall: `npm run build-options`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the atom-beautify@0.29.16 postinstall script 'npm run build-options'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the atom-beautify package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     npm run build-options
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs atom-beautify
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls atom-beautify
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     C:\Users\BJARNE~1\AppData\Local\Temp\apm-install-dir-117011-11520-wea9ix\npm-debug.log
npm ERR! code 1
```

**Comment by prettydiff** [comment]
These look like two separate issues.  It looks like @dathanasakoglou the problem is a timeout with regards to some address over HTTPS, which could be a certificate or proxy problem.  I am not sure why it is trying to make that request or even what that address is.  It looks like @BlackSkorpio the problem is a script failure that either crashed or performed a process.exit(1) internally.

**Comment by ChristopherParo** [comment]
@prettydiff mine looks similar to blackskorpio's any suggestions on how to fix it?

Installing “atom-beautify@0.29.16” failed.Hide output…
INFO: Could not find files for the given pattern(s).

The system cannot find the file specified.


> atom-beautify@0.29.16 postinstall C:\Users\CHRIST~1\AppData\Local\Temp\apm-install-dir-117011-4028-145tb6l\node_modules\atom-beautify
> npm run build-options

C:\Users\CHRIST~1\AppData\Local\Temp\apm-install-dir-117011-4028-145tb6l
`-- (empty)


npm WARN deprecated node-uuid@1.4.7: use uuid module instead
npm WARN deprecated minimatch@0.2.12: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated node-uuid@1.4.0: use uuid module instead
npm WARN deprecated minimatch@2.0.10: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
'C:\Users\Christopher' is not recognized as an internal or external command,
operable program or batch file.
npm WARN enoent ENOENT: no such file or directory, open 'C:\Users\CHRIST~1\AppData\Local\Temp\apm-install-dir-117011-4028-145tb6l\package.json'
npm WARN apm-install-dir-117011-4028-145tb6l No description
npm WARN apm-install-dir-117011-4028-145tb6l No repository field.
npm WARN apm-install-dir-117011-4028-145tb6l No README data
npm WARN apm-install-dir-117011-4028-145tb6l No license field.
npm ERR! Windows_NT 10.0.14393
npm ERR! argv "C:\\Users\\Christopher Paro\\AppData\\Local\\atom\\app-1.12.9\\resources\\app\\apm\\bin\\node.exe" "C:\\Users\\Christopher Paro\\AppData\\Local\\atom\\app-1.12.9\\resources\\app\\apm\\node_modules\\npm\\bin\\npm-cli.js" "--globalconfig" "C:\\Users\\Christopher Paro\\.atom\\.apm\\.apmrc" "--userconfig" "C:\\Users\\Christopher Paro\\.atom\\.apmrc" "install" "C:\\Users\\CHRIST~1\\AppData\\Local\\Temp\\d-117011-4028-1azwpcs\\package.tgz" "--runtime=electron" "--target=1.3.13" "--arch=ia32" "--global-style"
npm ERR! node v4.4.5
npm ERR! npm  v3.10.5
npm ERR! code ELIFECYCLE

npm ERR! atom-beautify@0.29.16 postinstall: `npm run build-options`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the atom-beautify@0.29.16 postinstall script 'npm run build-options'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the atom-beautify package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     npm run build-options
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs atom-beautify
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls atom-beautify
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     C:\Users\CHRIST~1\AppData\Local\Temp\apm-install-dir-117011-4028-145tb6l\npm-debug.log
npm ERR! code 1

**Comment by hsiktas** [comment]
My issue is similar and it seems like it can't handle the extra space in my profile's path.

**Comment by prettydiff** [comment]
It does seem space related after looking at the following:

`C:\Users\CHRIST1\AppData\Local\Temp\d-117011-4028-1azwpcs\package.tgz`

Where the correct path is probably

`C:\Users\Christopher Paro\AppData\Local\Temp\d-117011-4028-1azwpcs\package.tgz`

This could also be the cause of defect for @BlackSkorpio as well, but probably not for @dathanasakoglou 

@Glavin001 Have you seen this before and do you think its an NPM failure?

**Comment by banikojp** [comment]
I'm having a same issue.
not proxy, and no using spaces in any PATH .

I tried update first , but I had Failure. so I installed after uninstalled.
please suggestion.

> atom-beautify@0.29.16 postinstall /tmp/apm-install-dir-117013-19407-z87sr8/node_modules/atom-beautify
> npm run build-options


npm WARN deprecated node-uuid@1.4.7: use uuid module instead
npm WARN engine remark@6.2.0: wanted: {"node":">=0.11.0"} (current: {"node":"0.10.40","npm":"2.13.3"})
npm WARN engine typescript-formatter@2.3.0: wanted: {"node":">= 4.2.0"} (current: {"node":"0.10.40","npm":"2.13.3"})
npm WARN engine unified@5.1.0: wanted: {"node":">=0.11.0"} (current: {"node":"0.10.40","npm":"2.13.3"})
npm WARN engine remark-stringify@2.3.0: wanted: {"node":">=0.11.0"} (current: {"node":"0.10.40","npm":"2.13.3"})
npm WARN engine remark-parse@2.2.0: wanted: {"node":">=0.11.0"} (current: {"node":"0.10.40","npm":"2.13.3"})
npm WARN engine trough@1.0.0: wanted: {"node":">=0.11.0"} (current: {"node":"0.10.40","npm":"2.13.3"})
npm WARN engine request@2.79.0: wanted: {"node":">= 4"} (current: {"node":"0.10.40","npm":"2.13.3"})
npm WARN engine deep-extend@0.4.1: wanted: {"node":">=0.12.0","iojs":">=1.0.0"} (current: {"node":"0.10.40","npm":"2.13.3"})
npm WARN engine glob-expand@0.2.1: wanted: {"node":">0.10"} (current: {"node":"0.10.40","npm":"2.13.3"})
npm WARN engine markdown-table@1.0.0: wanted: {"node":">=0.11.0"} (current: {"node":"0.10.40","npm":"2.13.3"})
npm WARN deprecated minimatch@0.2.12: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated node-uuid@1.4.0: use uuid module instead
npm WARN deprecated minimatch@2.0.10: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN engine form-data@2.1.2: wanted: {"node":">= 0.12"} (current: {"node":"0.10.40","npm":"2.13.3"})
npm WARN engine array-iterate@1.1.0: wanted: {"node":">=0.11.0"} (current: {"node":"0.10.40","npm":"2.13.3"})
sh: 1: npm: not found
npm ERR! Linux 4.4.0-59-generic
npm ERR! argv "/usr/share/atom/resources/app/apm/bin/node" "/usr/share/atom/resources/app/apm/node_modules/npm/bin/npm-cli.js" "--globalconfig" "/home/yoshimura/.atom/.apm/.apmrc" "--userconfig" "/home/yoshimura/.atom/.apmrc" "install" "/tmp/d-117013-19407-yv4o3n/package.tgz" "--target=0.37.8" "--arch=x64"
npm ERR! node v0.10.40
npm ERR! npm  v2.13.3
npm ERR! file sh
npm ERR! code ELIFECYCLE
npm ERR! errno ENOENT
npm ERR! syscall spawn

npm ERR! atom-beautify@0.29.16 postinstall: `npm run build-options`
npm ERR! spawn ENOENT
npm ERR! 
npm ERR! Failed at the atom-beautify@0.29.16 postinstall script 'npm run build-options'.
npm ERR! This is most likely a problem with the atom-beautify package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     npm run build-options
npm ERR! You can get their info via:
npm ERR!     npm owner ls atom-beautify
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /tmp/apm-install-dir-117013-19407-z87sr8/npm-debug.log

**Comment by riccardo1991** [comment]
same here:
```
> atom-beautify@0.29.16 postinstall /private/var/folders/ck/z06msp7s7sd858_p9dp_wywh0000gn/T/apm-install-dir-117013-52438-cr60yv/node_modules/atom-beautify
> npm run build-options


npm WARN deprecated node-uuid@1.4.7: use uuid module instead
npm WARN engine remark@6.2.0: wanted: {"node":">=0.11.0"} (current: {"node":"0.10.40","npm":"2.13.3"})
npm WARN engine typescript-formatter@2.3.0: wanted: {"node":">= 4.2.0"} (current: {"node":"0.10.40","npm":"2.13.3"})
npm WARN engine remark-stringify@2.3.0: wanted: {"node":">=0.11.0"} (current: {"node":"0.10.40","npm":"2.13.3"})
npm WARN engine unified@5.1.0: wanted: {"node":">=0.11.0"} (current: {"node":"0.10.40","npm":"2.13.3"})
npm WARN engine remark-parse@2.2.0: wanted: {"node":">=0.11.0"} (current: {"node":"0.10.40","npm":"2.13.3"})
npm WARN engine trough@1.0.0: wanted: {"node":">=0.11.0"} (current: {"node":"0.10.40","npm":"2.13.3"})
npm WARN engine request@2.79.0: wanted: {"node":">= 4"} (current: {"node":"0.10.40","npm":"2.13.3"})
npm WARN engine deep-extend@0.4.1: wanted: {"node":">=0.12.0","iojs":">=1.0.0"} (current: {"node":"0.10.40","npm":"2.13.3"})
npm WARN engine markdown-table@1.0.0: wanted: {"node":">=0.11.0"} (current: {"node":"0.10.40","npm":"2.13.3"})
npm WARN engine glob-expand@0.2.1: wanted: {"node":">0.10"} (current: {"node":"0.10.40","npm":"2.13.3"})
npm WARN deprecated minimatch@0.2.12: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN engine array-iterate@1.1.0: wanted: {"node":">=0.11.0"} (current: {"node":"0.10.40","npm":"2.13.3"})
npm WARN deprecated node-uuid@1.4.0: use uuid module instead
npm WARN deprecated minimatch@2.0.10: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN engine form-data@2.1.2: wanted: {"node":">= 0.12"} (current: {"node":"0.10.40","npm":"2.13.3"})
sh: npm: command not found
npm ERR! Darwin 16.3.0
npm ERR! argv "/Applications/Atom.app/Contents/Resources/app/apm/bin/node" "/Applications/Atom.app/Contents/Resources/app/apm/node_modules/npm/bin/npm-cli.js" "--globalconfig" "/Users/Riccardo/.atom/.apm/.apmrc" "--userconfig" "/Users/Riccardo/.atom/.apmrc" "install" "/private/var/folders/ck/z06msp7s7sd858_p9dp_wywh0000gn/T/d-117013-52438-1rlk797/package.tgz" "--target=0.37.8" "--arch=x64"
npm ERR! node v0.10.40
npm ERR! npm  v2.13.3
npm ERR! file sh
npm ERR! code ELIFECYCLE
npm ERR! errno ENOENT
npm ERR! syscall spawn

npm ERR! atom-beautify@0.29.16 postinstall: `npm run build-options`
npm ERR! spawn ENOENT
npm ERR! 
npm ERR! Failed at the atom-beautify@0.29.16 postinstall script 'npm run build-options'.
npm ERR! This is most likely a problem with the atom-beautify package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     npm run build-options
npm ERR! You can get their info via:
npm ERR!     npm owner ls atom-beautify
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /private/var/folders/ck/z06msp7s7sd858_p9dp_wywh0000gn/T/apm-install-dir-117013-52438-cr60yv/npm-debug.log
```

**Comment by Glavin001** [comment]
>  it can't handle the extra space in my profile's path.

This a duplicate issue: https://github.com/Glavin001/atom-beautify/issues/1434
Which is actually an `apm` bug: https://github.com/atom/apm/issues/666
Which has been fixed with https://github.com/atom/apm/commit/e8b28ce31dc55890ca978ebe889a3de945d7cd99

I will leave this issue open as it appears this installation issue is not the space issue. However, anyone coming here because of a space in their path, see the issue at `apm`.

**Comment by Glavin001** [comment]
That being said, it is unlikely Atom-Beautify will be able to resolve these installation issues. You all should really contact `apm` for installation issues.

The only thing Atom-Beautify does which may be different from your other packages is utilize the `postinstall` script in `package.json` which is allowed and is meant to be supported by `apm` and is supported by `npm`: https://github.com/Glavin001/atom-beautify/blob/master/package.json#L302

> sh: npm: command not found

Your logs look like a different issue entirely, @riccardo1991 and @banikojp. You should contact `apm` team, as `npm` should be available within the `postinstall` script.

> npm ERR! network connect ETIMEDOUT 151.101.112.162:443

@dathanasakoglou this is something you should contact `apm` team about. Atom-Beautify publishes to `npm` and `apm`, both of which are officially supported registries. Any network issues would therefore be their domain as they are out of our control in Atom-Beautify at that stage.

I am closing this issue as all of the comments above should be forwarded to `apm`. Please let me know if you have any questions and please link back to this Issue when you create your new issue for `apm`. Feel free to CC on your new `apm` issues as well if there are any questions. Thank you.

**Comment by Glavin001** [comment]
I've merged https://github.com/Glavin001/atom-beautify/pull/1598 which should resolve these issues.

Published to v0.29.21.
